src/main/kotlin/com/example/iot_management_device/service/user/UserServiceImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.iot_management_device.service.user
+
+import com.example.iot_management_device.dto.user.request.UserRegistrationRequestDto
+import com.example.iot_management_device.dto.user.response.UserResponseDto
+import com.example.iot_management_device.dto.user.request.UserUpdateRequestDto
+import com.example.iot_management_device.exception.EntityNotFoundException
+import com.example.iot_management_device.mapper.UserMapper
+import com.example.iot_management_device.model.Role
+import com.example.iot_management_device.repository.DeviceRepository
+import com.example.iot_management_device.repository.RoleRepository
+import com.example.iot_management_device.repository.UserRepository
+import jakarta.transaction.Transactional
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Service
+
+@Service
+class UserServiceImpl(
+    private val userRepository: UserRepository,
+    private val userMapper: UserMapper,
+    private val passwordEncoder: PasswordEncoder,
+    private val roleRepository: RoleRepository,
+    private val deviceRepository: DeviceRepository
+) : UserService {
+
+    override fun register(requestDto: UserRegistrationRequestDto): UserResponseDto {
+        requestDto.userPassword = passwordEncoder.encode(requestDto.userPassword)
+        val userRole = roleRepository.findByRoleName(Role.RoleName.USER)
```

Comment:
Do we realy need send request in db for same user role object every time we register an user? - Maybe we should create constant or bean object in config